src/rest_vol_object.c

```diff
@@ -726,3 +1275,799 @@ RV_get_object_info_callback(char *HTTP_response, void *callback_data_in, void *c
+            FUNC_GOTO_ERROR(H5E_LINK, H5E_CANTGET, FAIL, "retrieval of link creation time failed");
+
+        if (!YAJL_IS_DOUBLE(link_field_obj))
+            FUNC_GOTO_ERROR(H5E_LINK, H5E_BADVALUE, FAIL, "returned link creation time is not a double");
+
+        table[i].crt_time = YAJL_GET_DOUBLE(link_field_obj);
+
+        /* Process the JSON for the current link and fill out a H5L_info2_t struct for it */
+
+        /* Find the beginning and end of the JSON section for this link */
+        if (NULL == (link_section_start = strstr(link_section_start, "{")))
+            FUNC_GOTO_ERROR(H5E_LINK, H5E_PARSEERROR, FAIL,
+                            "can't find start of current link's JSON section");
+
+        /* Continue forward through the string buffer character-by-character until the end of this JSON
+         * object section is found.
+         */
+        FIND_JSON_SECTION_END(link_section_start, link_section_end, H5E_LINK, FAIL);
+
+        /* Since it is not important if we destroy the contents of the HTTP response buffer,
+         * NULL terminators will be placed in the buffer strategically at the end of each link
+         * subsection (in order to "extract out" that subsection) corresponding to each individual
+         * link, and pass it to the "get link info" callback.
+         */
+        *link_section_end = '\0';
+
+        /* Fill out a H5L_info2_t struct for this link */
+        if (RV_parse_response(link_section_start, NULL, &table[i].link_info, RV_get_link_info_callback) < 0)
+            FUNC_GOTO_ERROR(H5E_LINK, H5E_CANTGET, FAIL, "couldn't get link info");
+
+        /* Store id for object in table */
+        switch (table[i].link_info.type) {
+            case H5L_TYPE_HARD:
+                H5VL_loc_by_idx_t loc_params;
+                loc_params.order = H5_ITER_INC;
+                loc_params.n     = 0;
+
+                /* Getting the link URI from the manually-manipulated links section requires a unique callback
+                 */
+                char *link_id_ptr = link_section_start;
+
+                char recent1 = '\0';
+                char recent2 = '\0';
+
+                while (link_id_ptr <= link_section_end) {
+                    char curr = *link_id_ptr;
+                    recent1   = recent2;
+                    recent2   = curr;
+
+                    if (recent1 == 'i' && recent2 == 'd') {
+                        break;
+                    }
+
+                    link_id_ptr++;
+                }
+
+                /* Move the pointer from the "d" in "id" the start of the id itself in the response. */
+                link_id_ptr += 5;
+
+                char *id_end_ptr = strstr(link_id_ptr, "\"");
+
+                size_t id_len = (size_t)(id_end_ptr - link_id_ptr);
+
+                memcpy(table[i].object_URI, link_id_ptr, id_len);
+                table[i].object_URI[id_len] = '\0';
```

Comment:
Is the above not something that can be done with YAJL before the NUL terminator is placed into the link subsection?